smartsim/experiment.py

```diff
@@ -797,6 +796,89 @@ def reconnect_orchestrator(self, checkpoint: str) -> Orchestrator:
             logger.error(e)
             raise
 
+    def preview(
+        self,
+        output_format: t.Optional[t.Literal["html"]] = None,
+        output_filename: t.Optional[str] = None,
+        verbosity_level: t.Literal["info", "debug", "developer"] = "info",
+    ) -> None:
+        """Preview entity information prior to launch. This method
+        aggregates multiple pieces of information to give users insight
+        into what and how entities will be launched.  Any instance of
+        ``Model``, ``Ensemble``, or ``Orchestrator`` created by the
+        Experiment can be passed as an argument to the preview method.
+        :param output_format: Set output destination. The possible accepted
+        output formats are `json`, `xml`, `html`, `plain_text`, `color_text`.
+        A filename is required if an output format is specified. If no output
+        format is set, the preview will be output to stdout. Defaults to None.
+        :type output_type: str
+        :param output_filename: Specify name of path to write preview data to.
+        Only needed when an output format has been specified. Defaults to None.
+        :type output_filename: str
+        :param verbosity level: the verbosity level.
+        info: Display User defined fields and entities
+        debug: Display user defined field and entities and auto generated
+        fields.
+        developer: Display user defined field and entities, auto generated
+        fields, and run commands.
+        Defaults to info.
+        :type verbosity_level: str
+        """
+
+        rendered_preview = previewrenderer.render(self, verbosity_level)
+
+        self._check_verbosity_level(verbosity_level)
+
+        if output_format:
+            self._check_output_format(output_format)
+        else:
+            if output_filename:
+                raise ValueError(
+                    "Output filename is only a valid parameter when an output \
+format is specified"
+                )
+
+        if output_format == "html":
+            if not output_filename:
+                raise ValueError(
+                    "An output filename is required when an output format is set."
+                )
+            previewrenderer.preview_to_file(rendered_preview, output_filename)
+
+        logger.info(rendered_preview)
+
+    @staticmethod
+    def _check_output_format(output_format: t.Literal["html"]) -> None:
+        if not output_format.startswith("html"):
+            raise ValueError("The only valid currently available is html")
+
+    @staticmethod
+    def _check_verbosity_level(
+        verbosity_level: t.Literal["info", "debug", "developer"]
+    ) -> None:
+        """
+        Check verbosity_level
+        """
+        if verbosity_level == "debug":
```

Comment:
Are we duplicating the verbosity levels of the logging system but using them inside of the templates? If so, can we use something that already exists in `smartsim.log` instead of having something in preview that also couples tightly to log levels?